hpedockerplugin/volume_manager.py

```diff
@@ -1515,14 +1604,21 @@ def _mount_volume(driver):
                  'msg': 'Removing mount directory: %s...' % mount_dir})
 
             # mount the directory
-            fileutil.mount_dir(path.path, mount_dir)
-            LOG.debug('Device: %(path)s successfully mounted on %(mount)s',
-                      {'path': path.path, 'mount': mount_dir})
+            try:
+                fileutil.mount_dir(path.path, mount_dir)
+                LOG.debug('Device: %(path)s successfully mounted on %(mount)s',
+                          {'path': path.path, 'mount': mount_dir})
 
-            undo_steps.append(
-                {'undo_func': fileutil.umount_dir,
-                 'params': mount_dir,
-                 'msg': 'Unmounting directory: %s...' % mount_dir})
+                undo_steps.append(
+                    {'undo_func': fileutil.umount_dir,
+                     'params': mount_dir,
+                     'msg': 'Unmounting directory: %s...' % mount_dir})
+            except Exception as ex:
+                msg = "Mount volume failed: %s" % six.text_type(ex)
+                LOG.error(msg)
+                self._rollback()
```

Comment:
needs to be changed from self._rollback()  to self._rollback(undo_steps)

Reply:
This is an outdated code... looking at the refreshed PR should show the updated code